contracts/interfaces/IEverlongStrategy.sol

```diff
@@ -0,0 +1,147 @@
+    // ╰───────────────────────────────────────────────────────────────────────╯
+
+    /// @notice Contains the information needed to identify an open Hyperdrive
+    ///         position.
+    struct Position {
+        /// @notice Time when the position matures.
+        uint128 maturityTime;
+        /// @notice Amount of bonds in the position.
+        uint128 bondAmount;
+    }
+
+    /// @notice Configuration for how `tend()` will be performed.
+    struct TendConfig {
+        /// @notice Minimum amount of bonds to receive when opening a position.
+        uint256 minOutput;
+        /// @notice Minimum vault share price when opening a position.
+        uint256 minVaultSharePrice;
+        /// @notice Maximum amount of mature positions that can be closed.
+        /// @dev A value of zero indicates no limit.
+        uint256 positionClosureLimit;
+        /// @notice Passed to hyperdrive `openLong()` and `closeLong()`.
+        bytes extraData;
+    }
+
+    /// @notice Parameters to specify how a rebalance will be performed.
+    struct RebalanceOptions {
+        /// @notice Limit on the amount of idle to spend on a new position.
+        /// @dev A value of zero indicates no limit.
+        uint256 spendingLimit;
+        /// @notice Minimum amount of bonds to receive when opening a position.
+        uint256 minOutput;
+        /// @notice Minimum vault share price when opening a position.
+        uint256 minVaultSharePrice;
+        /// @notice Maximum amount of mature positions that can be closed.
+        /// @dev A value of zero indicates no limit.
+        uint256 positionClosureLimit;
+        /// @notice Passed to hyperdrive `openLong()` and `closeLong()`.
+        bytes extraData;
+    }
+
+    // ╭───────────────────────────────────────────────────────────────────────╮
+    // │                                SETTERS                                │
+    // ╰───────────────────────────────────────────────────────────────────────╯
+
+    /// @notice Sets the minimum number of bonds to receive when opening a long.
+    /// @param _minOutput Minimum number of bonds to receive when opening a long.
+    function setMinOutput(uint256 _minOutput) external;
+
+    /// @notice Sets the minimum vault share price when opening a long.
+    /// @param _minVaultSharePrice Minimum vault share price when opening a long.
+    function setMinVaultSharePrice(uint256 _minVaultSharePrice) external;
+
+    /// @notice Sets the max amount of mature positions to close at a time.
+    /// @param _positionClosureLimit Max amount of mature positions to close at
+    ///        a time.
+    function setPositionClosureLimit(uint256 _positionClosureLimit) external;
+
+    /// @notice Sets the extra data to pass to hyperdrive when opening/closing
+    ///         longs.
+    /// @param _extraData Extra data to pass to hyperdrive when opening/closing
+    ///         longs.
+    function setExtraData(bytes memory _extraData) external;
```

Comment:
I remember talking about this. This seems fine for L2s but will get expensive on L1. Is this how all yearn vaults are managed, or do some of them handle this differently?

Reply:
From the few that I've looked at their source, they've used storage-based configuration like above...

FWIW I don't expect these to have to change much since they're mainly for edge cases.

I'll ask the yearn team about their keeper infra setup and how difficult it'd be to accommodate a non-standard tend function.

Reply:
Won't `minOutput` have to change every time we rebalance? It's a function of both the amount of base that is being used to buy bonds and the fixed rate.

Reply:
as discussed offline we'll implement the parameterized tend function in a later PR which will make this more relevant/useful